### PR TITLE
feat: number and epoch types, IntervalValue list parsing

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: src/czml3/types.py [hunk 1]
```diff
@@ -343,9 +343,12 @@ class IntervalValue(BaseCZMLObject):
     def to_json(self):
         obj_dict = {"interval": TimeInterval(start=self._start, end=self._end)}
 
-        try:
+        if hasattr(self._value, "to_json"):
```
Comment (Stoops-ML) [inline src/czml3/types.py]: Replace with `if isinstance(self._value, BaseCZMLObject)`

**Review by Stoops-ML — CHANGES_REQUESTED** [review]
Great work!

File: src/czml3/types.py (hunk 1 above)
Comment (vishwa2710) [inline src/czml3/types.py]:
```suggestion
        if isinstance(self._value, BaseCZMLObject)
```

**Comment by vishwa2710** [comment]
@Stoops-ML done!

File: src/czml3/types.py [hunk 2]
```diff
@@ -343,7 +343,7 @@ class IntervalValue(BaseCZMLObject):
     def to_json(self):
         obj_dict = {"interval": TimeInterval(start=self._start, end=self._end)}
 
-        if hasattr(self._value, "to_json"):
+        if isinstance(self._value, BaseCZMLObject)
```
Comment (Stoops-ML) [inline src/czml3/types.py]: Missing colon

**Review by Stoops-ML — COMMENTED** [review]
Small fix required.

File: src/czml3/types.py [hunk 3]
```diff
@@ -343,9 +343,12 @@ class IntervalValue(BaseCZMLObject):
     def to_json(self):
         obj_dict = {"interval": TimeInterval(start=self._start, end=self._end)}
 
-        try:
+        if isinstance(self._value, BaseCZMLObject)
```
Comment (vishwa2710) [inline src/czml3/types.py]:
```suggestion
        if isinstance(self._value, BaseCZMLObject):
```

**Comment by vishwa2710** [comment]
🤦 thanks! This should be good.

**Comment by Stoops-ML** [comment]
The test `test_interval_value()` failed because the class `CustomValue` is not a subclass of `BaseCZMLObject`.

**Comment by vishwa2710** [comment]
Ah sorry for the mess. I wasn't running the tests locally as the setup is not containerized. I went through the motions and I've tested this locally. It should pass now 💯

**Comment by vishwa2710** [comment]
Would we be opposed to containerizing this repo and adding a makefile with some simple commands for testing, formatting, etc.?

**Comment by Stoops-ML** [comment]
Need to run `ruff format` and `ruff check --fix` to format the code.

**Comment by Stoops-ML** [comment]
> Would we be opposed to containerizing this repo and adding a makefile with some simple commands for testing, formatting, etc.?

Open an issue on this. I'd like to hear what others think about this.

**Comment by vishwa2710** [comment]
Checked with ruff 👍

**Review by Stoops-ML — APPROVED** [review]

**Pull request merged**